src/main/kotlin/dev/arbjerg/lavalink/internal/ReconnectTask.kt

```diff
@@ -14,8 +13,7 @@ class ReconnectTask(val lavalink: LavalinkClient) : Runnable {
         try {
             sockets.forEach { ws ->
                 if (
-                    !ws.socket!!.isOpen &&
-                    ws.socket!!.state != WebSocketState.CONNECTING &&
```

Comment:
afaik the okhttp client doesn't expose a connecting state

Reply:
I checked, it does not. Not going to merge it because of that.

Reply:
That state is identical to the new `open` var

WebSocket.Factory states

> Creates a new web socket and immediately returns it. Creating a web socket initiates an asynchronous process to connect the socket. **Once that succeeds or fails, listener will be notified**. The caller must either close or cancel the returned web socket when it is no longer in use.

Since the only thing that check actually cares about is whether the connection is established or not, once onOpen is called and if it closes onClosed is called